target_snowflake/__init__.py

```diff
@@ -130,45 +135,53 @@ def persist_lines(config, lines, information_schema_cache=None):
                 validators[stream].validate(float_to_decimal(o['record']))
             except Exception as ex:
                 if type(ex).__name__ == "InvalidOperation":
-                    logger.error("Data validation failed and cannot load to destination. RECORD: {}\n'multipleOf' validations that allows long precisions are not supported (i.e. with 15 digits or more). Try removing 'multipleOf' methods from JSON schema."
-                    .format(o['record']))
+                    logger.error(
+                        "Data validation failed and cannot load to destination. RECORD: {}\n'multipleOf' validations that allows long precisions are not supported (i.e. with 15 digits or more). Try removing 'multipleOf' methods from JSON schema."
+                            .format(o['record']))
                     raise ex
 
             primary_key_string = stream_to_sync[stream].record_primary_key_string(o['record'])
             if not primary_key_string:
-                primary_key_string = 'RID-{}'.format(row_count[stream])
+                primary_key_string = 'RID-{}'.format(total_row_count[stream])
 
             if stream not in records_to_load:
                 records_to_load[stream] = {}
 
+            # If primary key doesn't exist in stream records
+            # it means we're encountering a new record thus increment counts
+            if primary_key_string not in records_to_load[stream]:
+                row_count[stream] += 1
+                total_row_count[stream] += 1
+
+            # append record
             if config.get('add_metadata_columns') or config.get('hard_delete'):
                 records_to_load[stream][primary_key_string] = add_metadata_values_to_record(o, stream_to_sync[stream])
             else:
                 records_to_load[stream][primary_key_string] = o['record']
 
-            row_count[stream] = len(records_to_load[stream])
-
             if row_count[stream] >= batch_size_rows:
```

Comment:
`batch_size_rows` in the integration tests always the default `10000` and seems like we never really test this condition with the current tests.

I added `config['batch_size_rows'] = 2` to `tests/utils.py` and `test_logical_streams_from_pg` test fails and inserting some not expected records to `logical1_table2`